data/metric/metrics.go

```diff
@@ -177,6 +206,62 @@ func NewCumulativeRate(timestamp time.Time, name string, value float64) (Metric,
 	}, nil
 }
 
+// NewPrometheusHistogram creates a new metric structurally similar to a Prometheus histogram
+func NewPrometheusHistogram(timestamp time.Time, name string, sampleCount uint64, sampleSum float64) (*PrometheusHistogram, error) {
+	return &PrometheusHistogram{
+		metricBase: metricBase{
+			Timestamp:  timestamp.Unix(),
+			Name:       name,
+			Type:       SourcesTypeToName[PROMETHEUS_HISTOGRAM],
+			Dimensions: Dimensions{},
+		},
+		// TODO make sure we make we don't allow NaN
+		SampleCount: &sampleCount,
```

Comment:
should we add a check also in the New function?

Reply:
no need. this comment is obsolete